app/models/components/course/materials_ability_component.rb

```diff
@@ -63,4 +63,22 @@ def concrete_folder_hash
     # Linked folders(folders with owners) are not manageable
     { owner_id: nil }
   end
+
+  # Involve Course#advance_start_at_duration when calculating the start_at time.
+  def opened_material_hashes
+    max_start_at = Time.zone.now
+    # Extend start_at time with self directed time from course settings.
+    max_start_at += (course.advance_start_at_duration || 0) if course
```

Comment:
I rather the code review process involve less changes, rather than rely only on the reviewers to catch missing pieces - ie. developers (new ones) could read the code and figure out some of these things on their own. 

Anyway, having this here is okay, so I think we can close this discussion.